src/main/java/nextstep/subway/application/LineService.java

```diff
@@ -0,0 +1,64 @@
+package nextstep.subway.application;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.LineRepository;
+import nextstep.subway.domain.Station;
+import nextstep.subway.domain.StationRepository;
+import nextstep.subway.dto.LineRequest;
+import nextstep.subway.dto.LineResponse;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+
+    private final LineRepository lineRepository;
+    private final StationRepository stationRepository;
+
+    public LineService(LineRepository lineRepository, StationRepository stationRepository) {
+        this.lineRepository = lineRepository;
+        this.stationRepository = stationRepository;
+    }
+
+    @Transactional
+    public LineResponse saveLine(LineRequest lineRequest) {
+        Station downStation = stationRepository.findById(lineRequest.getDownStationId()).orElse(null);
+        Station upStation = stationRepository.findById(lineRequest.getUpStationId()).orElse(null);
+
+        Line persistLine = lineRepository.save(lineRequest.toLine(downStation, upStation));
+        return LineResponse.of(persistLine);
+    }
+
+    public List<LineResponse> findAllLines() {
+        List<Line> lineList = lineRepository.findAll();
+
+        return lineList.stream()
+            .map(LineResponse::of)
+            .collect(Collectors.toList());
+    }
+
+    public LineResponse findLine(Long lineId) {
+        Line line = lineRepository.findById(lineId).orElse(null);
+        return LineResponse.of(line);
+    }
+
+    @Transactional
+    public void modifyLine(Long lineId, LineRequest lineRequest) {
+        Optional<Line> optionalLine = lineRepository.findById(lineId);
```

Comment:
```suggestion
       Line optionalLine = lineRepository.findById(lineId).orElseThrow(...);
```
`Optional` 을 활용해보시는건 어떨까요?
코드를 조금 더 깔끔하게 만들 수 있어요.